pkg/executor/cms/cmscontroller.go

```diff
@@ -0,0 +1,185 @@
+	nd "github.com/fission/fission/pkg/executor/newdeploy"
+	gpm "github.com/fission/fission/pkg/executor/poolmgr"
+)
+
+type (
+	ConfigSecretController struct {
+		logger *zap.Logger
+
+		configmapController cache.Controller
+		secretController    cache.Controller
+
+		fissionClient *crd.FissionClient
+	}
+)
+
+//MakeConfigSecretController makes a controller for configmaps and secrets which changes related functions
+func MakeConfigSecretController(logger *zap.Logger, fissionClient *crd.FissionClient,
+	kubernetesClient *kubernetes.Clientset, ndm *nd.NewDeploy, gpm *gpm.GenericPoolManager) *ConfigSecretController {
+	logger.Info("Creating ConfigMap & Secret Controller")
+	_, cmcontroller := initConfigmapController(logger, fissionClient, kubernetesClient, ndm, gpm)
+	_, scontroller := initSecretController(logger, fissionClient, kubernetesClient, ndm, gpm)
+	cmsController := &ConfigSecretController{
+		logger:              logger,
+		configmapController: cmcontroller,
+		secretController:    scontroller,
+		fissionClient:       fissionClient,
+	}
+	return cmsController
+}
+
+//Run runs the controllers for configmaps and secrets
+func (csController *ConfigSecretController) Run(ctx context.Context) {
+	go csController.configmapController.Run(ctx.Done())
+	go csController.secretController.Run(ctx.Done())
+}
+
+func initConfigmapController(logger *zap.Logger, fissionClient *crd.FissionClient,
+	kubernetesClient *kubernetes.Clientset, ndm *nd.NewDeploy, gpm *gpm.GenericPoolManager) (cache.Store, cache.Controller) {
+	resyncPeriod := 30 * time.Second
+	listWatch := cache.NewListWatchFromClient(kubernetesClient.Core().RESTClient(), "configmaps", metav1.NamespaceAll, fields.Everything())
+	store, controller := cache.NewInformer(listWatch, &apiv1.ConfigMap{}, resyncPeriod, cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) {},
+		DeleteFunc: func(obj interface{}) {},
+		UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+			oldCm := oldObj.(*apiv1.ConfigMap)
+			newCm := newObj.(*apiv1.ConfigMap)
+			if oldCm.ObjectMeta.ResourceVersion != newCm.ObjectMeta.ResourceVersion {
+				if newCm.ObjectMeta.Namespace != "kube-system" {
+					logger.Info("Configmap changed",
+						zap.String("configmap_name", newCm.ObjectMeta.Name),
+						zap.String("configmap_namespace", newCm.ObjectMeta.Namespace))
+
+				}
+
+				funcs := getConfigmapRelatedFuncs(logger, &newCm.ObjectMeta, fissionClient)
+				for _, f := range funcs {
+					// Poolmgr automatically picks up latest configmaps during specialization
+					if f.Spec.InvokeStrategy.ExecutionStrategy.ExecutorType == fv1.ExecutorTypeNewdeploy {
+						err := ndm.RecycleFuncPods(logger, f)
+						if err != nil {
+							logger.Error("Failed to recycle pods for function after configmap changed",
+								zap.Error(err),
+								zap.Any("function", f))
+						}
+					}
+
+					if f.Spec.InvokeStrategy.ExecutionStrategy.ExecutorType == fv1.ExecutorTypePoolmgr {
+						// Future implementation
```

Comment:
We need to document the different behavior of newdeploy/poolmgr when facing secret/configmap update in `https://docs.fission.io/usage/access-secret-cfgmap-in-function/` so that users won't confuse about this.